Controllers films users

Victorioussword opened the pull request.

Victorioussword committed:
1 11 2022 - переезд в новый репозиторий

Victorioussword committed:
1 11 2022 - переезд в новый репозиторий

Victorioussword committed:
1 11 2022 - все проверки пройдены

TyutterinYakov commented on pom.xml:
```diff
@@ -0,0 +1,61 @@
+	<artifactId>filmorate</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<name>filmorate</name>
+	<description>Demo project for Spring Boot</description>
```
не забываем исправить стандартное описание)

TyutterinYakov commented on pom.xml:
```diff
@@ -0,0 +1,61 @@
+	<artifactId>filmorate</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<name>filmorate</name>
+	<description>Demo project for Spring Boot</description>
```
не забываем исправить стандартное описание)

Victorioussword replied:
изменено поле Description

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,93 @@
+    private final static LocalDate FIRST_FILM_DATE = LocalDate.of(1895, 12, 28);
+    private final static int MAX_SIZE_DESCRIPTION = 200;
+
+    int id = 1;
```
Для соблюдения инкапсуляции поля класса должны быть приватными)

Victorioussword replied:
поле id сделано private

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,86 @@
+public class UserController {
+    private final static Logger log = LoggerFactory.getLogger(UserController.class);
+    private Map<Integer, User> users = new HashMap<>();
+    int id = 1;
```
Для соблюдения инкапсуляции поля класса должны быть приватными)

Victorioussword replied:
поле id сделано private

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,93 @@
+    int id = 1;
+
+    private Map<Integer, Film> films = new HashMap<>();
+    private final static Logger log = LoggerFactory.getLogger(FilmController.class);
```
Можно было бы произвести добавление логгера с помощью аннотации `@Slf4j` над классом)

Victorioussword replied:
Использована аннотация @Slf4j для создания логгера.

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,86 @@
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private final static Logger log = LoggerFactory.getLogger(UserController.class);
```
Можно было бы произвести добавление логгера с помощью аннотации `@Slf4j` над классом)

Victorioussword replied:
Использована аннотация @Slf4j для создания логгера.

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,86 @@
+
+    private User emptyNameCheck(@Valid @RequestBody User user) {
+        String name = user.getName();
+        if (name == null || name.isEmpty()) {
```
Хорошо было бы использовать не `isEmpty`, а `isBlank`, так как первый допускает строку, состоящую только из пробелов, что не совсем верно)

Victorioussword replied:
Использован isBlank()

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -0,0 +1,35 @@
+public class Film {
+
+    @Setter
+    @NotNull
```
Аннотация `@NotNull` не используется над примитивными типами)

Victorioussword replied:
Удалена аннотация @NotNull. Благодарю за пояснение. Эта информация очень помогла.

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -0,0 +1,35 @@
+    private int id;
+
+    @NotNull
+    @NotBlank
```
Аннотация `@NotNull` не требуется, так как проверка на null входит в `@NotBlank`

Victorioussword replied:
Удален @NotNull.

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -0,0 +1,35 @@
+    final private String name;
+
+    @NotNull
+    @NotBlank
```
Аннотация `@NotNull` не требуется, так как проверка на null входит в `@NotBlank`

Victorioussword replied:
Удален @NotNull.

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -0,0 +1,35 @@
+    final private LocalDate releaseDate;
+
+    @NotNull
+    @Positive
```
Аннотация `@NotNull` не используется над примитивными типами)

Victorioussword replied:
Удален NotNull.

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -0,0 +1,35 @@
+
+    @Setter
+    @NotNull
+    private int id;
```
Для хранения идентификатора лучше использовать целочисленный тип `long`, так как типа `int` может не хватить)

Victorioussword replied:
тип переменной установлен long

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,34 @@
+public class User {
+
+    @NotNull
+    private  int id;
```
Для хранения идентификатора лучше использовать целочисленный тип `long`, так как типа `int` может не хватить)

Victorioussword replied:
тип переменной установлен long

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,34 @@
+public class User {
+
+    @NotNull
+    private  int id;
```
Аннотация `@NotNull` не используется над примитивными типами)

Victorioussword replied:
@NotNull - удалено

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,34 @@
+    @NotBlank
+    private String name;
+
+    @NotNull
```
В данном случае можно было бы использовать аннотацию `@PastOrPresent`, чтобы избавиться от проверки в коде)

Victorioussword replied:
Использована аннотация @PastOrPresent

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,34 @@
+    private final String email;
+
+    @NotNull
+    @NotBlank
```
Аннотация `@NotNull` не требуется, так как проверка на null входит в `@NotBlank`

Victorioussword replied:
@NotNull удалено

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,34 @@
+    private final String login;
+
+    @NotNull
+    @NotBlank
```
Аннотации валидации следует исключить над именем, так как, если оно не будет передано - это вполне штатная ситуация. В таком случае имя == логин)

Victorioussword replied:
аннотации удалены

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,34 @@
+
+    @NotNull
+    @Email
+    @NotBlank
```
Аннотация `@NotBlank` здесь излишня, так как, если строка будет передана - она в любом случае будет проверена с помощью `@Email` на соответствие формату почты, но, она допускает null, поэтому в данном случае так же используется и `@NotNull`

Victorioussword replied:
@NotBlank удалено

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -0,0 +1,35 @@
+    final private String name;
+
+    @NotNull
+    @NotBlank
```
Здесь так же можно было бы использовать аннотацию `@Size`, чтобы ограничить размер текстового поля)

Victorioussword replied:
Использована аннотация @Size

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,86 @@
+        }
+    }
+
+    private void loginCheck(@Valid @RequestBody User user) {
```
Аннотации `@Valid @RequestBody` используются только в методе-обработчике) Здесь они не требуются)

Victorioussword replied:
из вспомогательных методов удалены аннотации @Valid @RequestBody

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,86 @@
+        }
+    }
+
+    private User emptyNameCheck(@Valid @RequestBody User user) {
```
Аннотации `@Valid @RequestBody` используются только в методе-обработчике) Здесь они не требуются)

Victorioussword replied:
из вспомогательных методов удалены аннотации @Valid @requestbody

TyutterinYakov reviewed (commented):
Добрый день! Файлы по типу mvnw, maven-wrapper.jar и тд хорошо было бы указать в .gitignore, так как в них нет необходимости для работы с проектом. Так же отметил, как можно упростить валидацию и отказаться от некоторых проверок в коде) Хорошего дня! Если возникнут вопросы - пиши в слак, буду рад помочь :)

Victorioussword committed:
2-11-2022 - все проверки пройдены
Исправлены замечания ревьюера

Victorioussword committed:
2-11-2022 - все проверки пройдены
Исправлены замечания ревьюера. попытка 2

Victorioussword commented:
Исправил замечания

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,93 @@
+        }
+    }
+
+    private void checkName(Film film) {
```
Проверка не требуется, так как она реализована с помощью аннотации `@NotBlank` в классе Film)

Victorioussword replied:
удален метод checkName.

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,93 @@
+
+
+
+    private void checkDuration(Film film) {
```
Метод не требуется, так как проверка на то, что значение является положительным, реализована с помощью аннотации `@Positive` над полем duration)

Victorioussword replied:
удален метод checkDuration

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,85 @@
+        }
+    }
+
+    private User emptyNameCheck( User user) {
```
Можно было бы сделать метод void, так как он принимает ссылку на объект, а значит изменения происходят именно в объекте класса User, следовательно ввод переменных в методах putUser и postUser не имеет смысла)

Victorioussword replied:
метод изменен

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,85 @@
+    private int id = 1;
+
+    @GetMapping
+    public Collection<User> getUsers() {
```
Так как методы не летают в воздухе, а относятся к контроллеру, который работает с сущностью пользователь, упоминание "User" в названии методов не требуется) Аналогично с теми, что ниже)

Victorioussword replied:
изменено название метода

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,93 @@
+
+
+    @GetMapping
+    public Collection<Film> getUsers() {
```
Так как методы не летают в воздухе, а относятся к контроллеру, который работает с сущностью фильм, упоминание "Film" в названии методов не требуется) Аналогично с теми, что ниже)
Но, в данном случае, копипаст сыграл злую шутку и здесь упоминается даже не фильм)

Victorioussword replied:
изменено название метода

TyutterinYakov reviewed (commented):
Добрый вечер! Отметил оставшиеся недочеты :) Хорошего вечера! Жду на финальное ревью)

Victorioussword committed:
3-11-2022 - все проверки пройдены
Исправлены замечания ревьюера.
- удален метод checkName();
- удален метод checkDuration();
- изменен метод emptyNameCheck();
- изменены названия методов get(), put(), post;

TyutterinYakov reviewed (approved):
Добрый вечер! Поздравляю с успешной сдачей спринта! Ты большой молодец :) Успехов в обучении! Доброй ночи и отличного утра :)
